libcxx/include/cmath

```diff
@@ -765,6 +766,54 @@ _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 _Tp __constexpr_scalbn(_Tp _
   return __builtin_scalbn(__x, __exp);
 }
 
+#if _LIBCPP_STD_VER >= 17
+
+/// \return the Hermite polynomial \f$ H_n(x) \f$
+/// \note The implementation is based on the recurrence formula
+/// \f[
+///   H_{n+1}(x) = 2x H_n(x) - 2n H_{n-1}
+/// \f]
+/// Press, William H., et al. Numerical recipes 3rd edition: The art of
+/// scientific computing. Cambridge university press, 2007, p. 183.
+template <class _Real>
+_LIBCPP_HIDE_FROM_ABI _Real __hermite(unsigned __n, _Real __x) {
+  if (std::isnan(__x))
+    return std::numeric_limits<_Real>::quiet_NaN();
+
+  _Real __H_0{1};
+  if (__n == 0)
+    return __H_0;
+
+  _Real __H_n_prev = __H_0;
+  _Real __H_n      = 2 * __x;
+  for (unsigned __i = 1; __i < __n; ++__i) {
+    _Real __H_n_next = 2 * (__x * __H_n - __i * __H_n_prev);
+    __H_n_prev       = __H_n;
+    __H_n            = __H_n_next;
+  }
+  return __H_n;
+}
+
+inline _LIBCPP_HIDE_FROM_ABI double hermite(unsigned __n, double __x) { return std::__hermite(__n, __x); }
```

Comment:
Not attached: should the mathematical special functions be in the dylib? I expect this will be quite a bit of code, and it would be nice to be consistent between the special functions.

Reply:
I cannot anticipate how much code that will be nor what the threshold is when we should be using dynamic libraries. Could we make that switch with a later integration, e.g. once the LOC exceed some threshold?

Reply:
Moving code into the dylib has quite a few consequences and may change design considerations, so I don't think we should wait until some later point. Unless we want to make the functions experimental for now.

Reply:
Ok, I will need to look into that. Can you point me in some direction? Documentation or other sample code?

Reply:
From a discussion with @mordante: 

* This function is quite small a probably does not need to go into the dylib.
* Large code or special template magic could be moved there.
* Concerning the other special math functions: I assume they probably will not contain too much code nor  any special template magic. 
* We still have the option in the future to move other special math functions into the dylib even though this one might stay in the header.